src/main/java/com/itechart/javalab/library/controller/command/impl/SearchBooksCommand.java

```diff
@@ -0,0 +1,80 @@
+package com.itechart.javalab.library.controller.command.impl;
+
+import com.itechart.javalab.library.controller.command.Command;
+import com.itechart.javalab.library.controller.util.JspPageName;
+import com.itechart.javalab.library.controller.util.BookSearchValidator;
+import com.itechart.javalab.library.model.Book;
+import com.itechart.javalab.library.service.BookService;
+import com.itechart.javalab.library.service.impl.DefaultBookService;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+public class SearchBooksCommand implements Command {
+
+    private static final String REQUEST_ATTRIBUTE_LIST_BOOKS = "books";
+    private static final String REQUEST_IS_AVAILABLE_VALUE = "isAvailableOnly";
+    private static final String REQUEST_BOOK_TITLE_SEARCH_PARAMETER = "bookTitle";
+    private static final String REQUEST_BOOK_AUTHOR_SEARCH_PARAMETER = "bookAuthor";
+    private static final String REQUEST_BOOK_GENRE_SEARCH_PARAMETER = "bookGenre";
+    private static final String REQUEST_BOOK_DESCRIPTION_SEARCH_PARAMETER = "bookDescription";
+
+    @Override
+    public void execute(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+
+        boolean isAvailableOnly = getFilterValue(request);
+        String bookTitleSearchParameter = request.getParameter(REQUEST_BOOK_TITLE_SEARCH_PARAMETER);
+        String bookAuthorSearchParameter = request.getParameter(REQUEST_BOOK_AUTHOR_SEARCH_PARAMETER);
+        String bookGenreSearchParameter = request.getParameter(REQUEST_BOOK_GENRE_SEARCH_PARAMETER);
+        String bookDescriptionSearchParameter = request.getParameter(REQUEST_BOOK_DESCRIPTION_SEARCH_PARAMETER);
+
+        if (BookSearchValidator.isAtLeastOneParameterNull(bookTitleSearchParameter,
+                bookAuthorSearchParameter,
+                bookGenreSearchParameter,
+                bookDescriptionSearchParameter)) {
+
+            request.getServletContext().getRequestDispatcher(JspPageName.SEARCH_PAGE).forward(request, response);
+
+        } else if (BookSearchValidator.isAllParametersEmpty(bookTitleSearchParameter,
+                bookAuthorSearchParameter,
+                bookGenreSearchParameter,
+                bookDescriptionSearchParameter)) {
+
+            response.setStatus(409);
```

Comment:
do not use numbers for return status code, use `javax.servlet.http.HttpServletResponse#SC_BAD_REQUEST`.
and why 409 (**CONFLICT**)? 400 (**BAD_REQUEST**) is more appropriate?